internal/config/config.go

```diff
@@ -75,16 +75,17 @@ type TUIConfig struct {
 
 // Config is the main configuration structure for the application.
 type Config struct {
-	Data         Data                              `json:"data"`
-	WorkingDir   string                            `json:"wd,omitempty"`
-	MCPServers   map[string]MCPServer              `json:"mcpServers,omitempty"`
-	Providers    map[models.ModelProvider]Provider `json:"providers,omitempty"`
-	LSP          map[string]LSPConfig              `json:"lsp,omitempty"`
-	Agents       map[AgentName]Agent               `json:"agents,omitempty"`
-	Debug        bool                              `json:"debug,omitempty"`
-	DebugLSP     bool                              `json:"debugLSP,omitempty"`
-	ContextPaths []string                          `json:"contextPaths,omitempty"`
-	TUI          TUIConfig                         `json:"tui"`
+	Data          Data                              `json:"data"`
+	WorkingDir    string                            `json:"wd,omitempty"`
+	MCPServers    map[string]MCPServer              `json:"mcpServers,omitempty"`
+	Providers     map[models.ModelProvider]Provider `json:"providers,omitempty"`
+	LSP           map[string]LSPConfig              `json:"lsp,omitempty"`
+	Agents        map[AgentName]Agent               `json:"agents,omitempty"`
+	Debug         bool                              `json:"debug,omitempty"`
+	DebugLSP      bool                              `json:"debugLSP,omitempty"`
+	ContextPaths  []string                          `json:"contextPaths,omitempty"`
+	TUI           TUIConfig                         `json:"tui"`
+	SetupComplete bool                              `json:"setupComplete,omit"`
```

Comment:
my main feedback is that i don't think we should have `setupComplete` as part of the config, we should just store that state using another mechanism. i think the current "init" feature stores a file in `.opencode`? i guess this is different in that it's more of a global setup than a project setup, but we could do something similar (store a file in the global config dir).

Reply:
I can just make this a global variable instead, and mutate it accordingly.

Reply:
Extracted the setup functionality into its own module and cleaned this all up via https://github.com/sst/opencode/pull/14/commits/98de5ba936848619fc763ec90e958d242f91260d